llama.cpp

```diff
@@ -2119,6 +2119,13 @@ struct llama_model_loader {
         }
 
         get_key(llm_kv(LLM_KV_GENERAL_ARCHITECTURE), arch_name, false);
+
+        // backwards compatibility with pre-#4552
+        // TODO: remove after Mar 2024
+        if (arch_name == "phi2") {
+            arch_name = "phi";
+        }
```

Comment:
@teleprint-me Could you give this a test and see if it solves backwards compatibility with "phi2"?

I'm a bit worried if we don't handle the old name we'll get lot's of complaints and issues

Reply:
Ah nvm, it's actually not going to work because there are other parameters like `phi2.context_length`

Reply:
@ggerganov Yeah, that's why I broke it. I wish I had caught this sooner, but I've been preoccupied with a bunch of other stuff and I'm multitasking the best I can. 

I actually started working on the conversion scripts too and I still have a bunch of other stuff, but this seemed like it needed attention sooner than later.